lib/phlex/csv.rb

```diff
@@ -1,64 +1,95 @@
 # frozen_string_literal: true
 
 class Phlex::CSV
-	FORMULA_PREFIXES = Set["=", "+", "-", "@", "\t", "\r"].freeze
-	SPACE_CHARACTERS = Set[" ", "\t", "\r"].freeze
+	FORMULA_PREFIXES_MAP = Array.new(128).tap do |map|
+		"=+-@\t\r".each_byte do |byte|
+			map[byte] = true
+		end
+	end.freeze
+
+	UNDEFINED = Object.new
 
 	def initialize(collection, delimiter: ",")
 		@collection = collection
 		@delimiter = delimiter
-		@_headers = []
-		@_current_row = []
-		@_current_column_index = 0
-		@_first = true
+		@_row_buffer = []
 	end
 
 	attr_reader :collection
 
 	def call(buffer = +"", context: nil)
-		unless escape_csv_injection? == true || escape_csv_injection? == false
-			raise <<~MESSAGE
-				You need to define escape_csv_injection? in #{self.class.name}, returning either `true` or `false`.
+		ensure_escape_csv_injection_configured!
 
-				CSV injection is a security vulnerability where malicious spreadsheet formulae are used to execute code or exfiltrate data when a CSV is opened in a spreadsheet program such as Microsoft Excel or Google Sheets.
+		strip_whitespace = trim_whitespace?
+		escape_csv_injection = escape_csv_injection?
+		delimiter = @delimiter
+		row_buffer = @_row_buffer
+		has_yielder = respond_to?(:yielder, true)
```

Comment:
Let’s do a warning here if you have a yielder. We don’t need a yielder and wrapper.